hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaRolloutExecutor.java

```diff
@@ -139,30 +138,33 @@ public JpaRolloutExecutor(final RolloutTargetGroupRepository rolloutTargetGroupR
     @Override
     public void execute(final Rollout rollout) {
         LOGGER.debug("handle rollout {}", rollout.getId());
-
-        switch (rollout.getStatus()) {
-        case CREATING:
-            handleCreateRollout((JpaRollout) rollout);
-            break;
-        case DELETING:
-            handleDeleteRollout((JpaRollout) rollout);
-            break;
-        case READY:
-            handleReadyRollout(rollout);
-            break;
-        case STARTING:
-            handleStartingRollout(rollout);
-            break;
-        case RUNNING:
-            handleRunningRollout((JpaRollout) rollout);
-            break;
-        default:
-            LOGGER.error("Rollout in status {} not supposed to be handled!", rollout.getStatus());
-            break;
+        try {
+            switch (rollout.getStatus()) {
+                case CREATING:
+                    handleCreateRollout((JpaRollout) rollout);
+                    break;
+                case DELETING:
+                    handleDeleteRollout((JpaRollout) rollout);
+                    break;
+                case READY:
+                    handleReadyRollout(rollout);
+                    break;
+                case STARTING:
+                    handleStartingRollout(rollout);
+                    break;
+                case RUNNING:
+                    handleRunningRollout((JpaRollout) rollout);
+                    break;
+                default:
+                    LOGGER.error("Rollout in status {} not supposed to be handled!", rollout.getStatus());
+                    break;
+            }
+        } catch (final TransactionExecutionException e) {
+            LOGGER.error("Caught exception during rollout execution", e);
```

Comment:
Was there a specific reason why you have decided to catch the exception on the top level as opposed to directly around the usage of `DeploymentHelper#runInNewTransaction`? Personally I prefer the error handling to reside as near as possible to the code that could throw it, however if you have strong arguments I'm alright with leaving it as is